tests/grafana_addon_test.go

```diff
@@ -45,8 +61,43 @@ func TestHelmGrafanaAddon(t *testing.T) {
 		log.Fatal(err)
 	}
 
-	waitUntilServicesAvailable(t, *helmOptions.KubectlOptions, []string{"grafana"})
+	waitUntilServicesAvailable(t, *helmOptions.KubectlOptions, []string{grafanaNamespaceAndAddonName})
+	waitUntilDeploymentsAvailable(t, *helmOptions.KubectlOptions, []string{grafanaNamespaceAndAddonName})
+
+	valuesFile, err := os.ReadFile("../addons/grafana/values.yaml")
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var yamlData GrafanaValues
 
+	err = yaml.Unmarshal(valuesFile, &yamlData)
+
+	targetPort := 3000
+	if err == nil && yamlData.GrafanaSource.Service.TargetPort != 0 {
```

Comment:
Would it make sense to fail in case we cannot unmarshal the values?

In theory that means something is wrong on the format and the test might cover that as well.